cmd/mistryd/server.go

```diff
@@ -379,34 +391,73 @@ func RebuildImages(cfg *Config, log *log.Logger, out io.Writer, projects []strin
 
 	ctx := context.Background()
 	for _, project := range projects {
+		start := time.Now()
 		log.Printf("Rebuilding %s...\n", project)
 		j, err := NewJob(project, types.Params{}, "", cfg)
 		if err != nil {
-			r.failed++
+			r.failed = append(r.failed, project)
 			log.Printf("Failed to instantiate %s job with error: %s\n", project, err)
 			if failOnImageError {
 				return r, err
 			}
 		} else {
-			err = j.BuildImage(ctx, cfg.UID, client, out, true, true)
-			if err != nil {
-				r.failed++
-				log.Printf("Failed to build %s job %s with error: %s\n", project, j.ID, err)
+			var buildErr error
+			if verbose {
+				// pipe image build logs to the logger
+				pr, pw := io.Pipe()
+				buildResult := make(chan error)
+
+				go func() {
+					err := j.BuildImage(ctx, cfg.UID, client, pw, true, true)
+					pw.Close()
+					buildResult <- err
+				}()
+
+				scanner := bufio.NewScanner(pr)
+				for scanner.Scan() {
+					log.Print(scanner.Text())
+				}
+				buildErr = <-buildResult
+			} else {
+				// discard image build logs
+				buildErr = j.BuildImage(ctx, cfg.UID, client, ioutil.Discard, true, true)
+			}
+
+			if buildErr != nil {
+				r.failed = append(r.failed, project)
+				log.Printf("Failed to build %s job %s with error: %s\n", project, j.ID, buildErr)
 				if failOnImageError {
-					return r, err
+					return r, buildErr
 				}
 			} else {
+				log.Printf("Rebuilt %s in %s\n", project, time.Now().Sub(start).Truncate(time.Millisecond))
 				r.successful++
 			}
 		}
 	}
-	r.pruneReport, err = client.ImagesPrune(ctx, dockerFilters.Args{})
+	r.prunedImages, r.prunedContainers, r.reclaimedSpace, err = dockerPruneUnused(ctx, client)
 	if err != nil {
 		return r, err
 	}
 	return r, nil
 }
 
+func dockerPruneUnused(ctx context.Context, c *docker.Client) (int, int, uint64, error) {
+	// TODO: we should look at filtering images and containers based on labels so that we
```

Comment:
Labels seem nice. Consider opening a new issue for adding mistry related labels and remove the `TODO`.